File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-0.cpp
class Solution {
public:
    vector<int> replaceElements(vector<int>& arr) {
        int n=arr.size();
        vector<int>x(n);
        int maxr=-1;

        
        for(int i=arr.size()-1;i>=0;--i){
            x[i]=maxr;
            maxr=max(arr[i],maxr);

        }
        
        return x;
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
         int r=0,l=0;
         int count=0;
         HashSet<Character>t=new HashSet<>();
         for(r=0;r<s.length();r++){
            while(t.contains(s.charAt(r))){
                t.remove(s.charAt(l));
                l++;
                
            }
            t.add(s.charAt(r));
            count=Math.max(count,r-l+1);
         }
         return count;
    }
}
